client/allocrunner/taskrunner/secrets_hook_test.go

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package taskrunner
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/nomad/ci"
+	"github.com/hashicorp/nomad/client/allocdir"
+	"github.com/hashicorp/nomad/client/allocrunner/interfaces"
+	trtesting "github.com/hashicorp/nomad/client/allocrunner/taskrunner/testing"
+	"github.com/hashicorp/nomad/client/config"
+	"github.com/hashicorp/nomad/client/taskenv"
+	"github.com/hashicorp/nomad/helper/bufconndialer"
+	"github.com/hashicorp/nomad/helper/testlog"
+	"github.com/hashicorp/nomad/nomad/mock"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/shoenig/test/must"
+)
+
+func TestSecretsHook_Prestart_Nomad(t *testing.T) {
+	ci.Parallel(t)
+
+	secretsResp := `
+	{
+	  "CreateIndex": 812,
+	  "CreateTime": 1750782609539170600,
+	  "Items": {
+	    "key2": "value2",
+	    "key1": "value1"
+	  },
+	  "ModifyIndex": 812,
+	  "ModifyTime": 1750782609539170600,
+	  "Namespace": "default",
+	  "Path": "testnomadvar"
+	}
+	`
+	count := 0 // CT expects a nomad index header that incremements, or else it continues polling
+	nomadServer := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I like that we're just running a mock server instead of making a bogus plugin. That way we're exercising it end-to-end.